### DISCUSSION TITLE
Is it possible to use rev() on a PyTupleIterator to achieve reverse iteration?

### DISCUSSION THREAD

**Discussion by AphrosDelos532** [PyO3/pyo3#3364] [Questions] [answered]
Here is my code,
```rust
    let args_iter = args.iter().rev();
```
Then i got,
```
error[E0277]: the trait bound `types::tuple::PyTupleIterator<'_>: DoubleEndedIterator` is not satisfied
    --> src/lib.rs:431:21
     |
431  |     let args_iter = args.iter().rev();
     |                     ^^^^^^^^^^^ --- required by a bound introduced by this call
     |                     |
     |                     the trait `DoubleEndedIterator` is not implemented for `types::tuple::PyTupleIterator<'_>`
     |
note: required by a bound in `rev` 
```

**Comment by adamreichold** [comment, marked as answer]
We have not yet implemented `DoubleEndedIterator` for `PyTupleIterator` which should however be possible. I will open a PR to fix this.

As workaround, maybe `PyTuple::as_slice` would work for you? Or in the worst case, using `PyTuple::get_item` to manually index into the tuple?

**Reply by AphrosDelos532** [reply]
I do this by collecting the elements into a Vec. However, I anticipate there might be some additional overhead. I'm glad to hear that reverse iterator will be supported soon. Thank you for your help!

**Reply by adamreichold** [reply]
https://github.com/PyO3/pyo3/pull/3366 is that PR.
